src/jdk.jdwp.agent/share/native/libdt_socket/socketTransport.c

```diff
@@ -460,6 +460,12 @@ parseAllowedMask(const char *buffer, int isIPv4, struct in6_addr *result) {
     int maxValue = isIPv4 ? 32 : 128;
     int i;
 
+    // on AIX we might get IPv6 detected (by getaddrinfo) as IPv4,
+    // so the default maxValue does not always work
+#if defined(_AIX)
+    maxValue = 128;
+#endif
+
```

Comment:
IIUC, this code would work for any port, but is only required for AIX. And since it is a hack to work around an AIX issue, you didn't want to subject all ports to it. The downside of this change is that it will cause the option parsing code to not properly produce an error if an IPv4 mask prefix between 32 and 128 is given.

Reply:
> The downside of this change is that it will cause the option parsing code to not properly produce an error if an IPv4 mask prefix between 32 and 128 is given.

Yes this is true; maybe we should not  unconditionally increase the maxValue on AIX .